### ISSUE TITLE
sql: non-tz timestamps don't roundtrip when session offset non-zero

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/cockroach#9266]
```
$ grep -F '[config]' cockroach-data/logs/cockroach.INFO
I160910 22:57:52.142813 1 util/log/clog.go:1019  [config] file created at: 2016/09/10 22:57:52
I160910 22:57:52.142813 1 util/log/clog.go:1019  [config] running on machine: JESSEs-MacBook-Pro
I160910 22:57:52.142813 1 util/log/clog.go:1019  [config] binary: CockroachDB beta-20160908 (darwin amd64, built 2016/09/08 15:37:08, go1.7)
I160910 22:57:52.142813 1 util/log/clog.go:1019  [config] arguments: [./cockroach start]
```

You can insert a timestamp with timezone offset into a `TIMESTAMP` column. It behaves the same as when the column is `TIMESTAMPTZ`.

**What did you do?**

~~~
root@:26257> CREATE TABLE test (a INT PRIMARY KEY, b TIMESTAMPTZ, c TIMESTAMP);
CREATE TABLE
root@:26257> INSERT INTO test VALUES (1, TIMESTAMPTZ '2016-03-26 10:10:10-05:00', TIMESTAMP '2016-03-26 10:10:10-05:00');
INSERT 1
root@:26257> select * from test;
+---+---------------------------------+---------------------------------+
| a |                b                |                c                |
+---+---------------------------------+---------------------------------+
| 1 | 2016-03-26 15:10:10 +0000 +0000 | 2016-03-26 15:10:10 +0000 +0000 |
+---+---------------------------------+---------------------------------+
(1 row)
~~~

**What did you expect to see?**

I expected it to not be possible to insert a timestamp with tz offset into a `timestamp` column.

**Comment by jseldess** [comment]
@dt, I think you worked on timestamps sometime recently, so perhaps you can explain this?

**Comment by dt** [comment]
So _that_ is actually the expected behavior, but in the process of looking at this, we uncovered that we're incorrectly applying the session offset to non-tz timestamps that do not specify an offset, meaning they don't round-trip. Given that they are the default, that's pretty bad. Our logic test on this only _writes_ with a zero session offset to we missed this when it was introduced in a refactor.

**Comment by dt** [comment]
Fixed by #9444
